packages/react/src/components/Masthead/MastheadLeftNav.js

```diff
@@ -86,23 +87,75 @@ const MastheadLeftNav = ({ navigation, isSideNavExpanded, platform }) => {
 /**
  * Loops through and renders a list of links for the side nav
  *
- * @param {Array} sections A list of links to be rendered
+ * @param {Array} sections A list of link sections to be rendered
  * @returns {object} JSX object
  */
-function renderNav(sections) {
-  const navItems = [];
+function renderNavSections(sections) {
+  const sectionItems = [];
   sections.forEach((section, i) => {
     section.menuItems.forEach((item, j) => {
-      navItems.push(
-        <SideNavMenuItem
-          href={item.url}
-          data-autoid={`${stablePrefix}--masthead__l0-sidenav--subnav-col${i}-item${j}`}
-          key={item.title}>
-          {item.title}
-        </SideNavMenuItem>
-      );
+      if (item.megapanelContent) {
+        sectionItems.push(
+          <SideNavMenu title={item.title} key={j}>
+            <SideNavMenuItem
+              onClick={event => event.preventDefault()}
+              className={`${prefix}--masthead__side-nav--submemu-back`}
+              data-autoid={`${stablePrefix}--masthead__l0-sidenav--subnav-back-${j}`}
+              isbackbutton="true"
+              key={i}>
+              <ChevronLeft20 />
+              Back
+            </SideNavMenuItem>
+
+            <SideNavLink
+              className={`${prefix}--masthead__side-nav--submemu-section-title`}
+              href={item.url}
+              data-autoid={`${stablePrefix}--masthead__l0-sidenav--nav-${i}`}
+              key={i}>
+              {item.title}
+              <div
+                className={`${prefix}--masthead__side-nav--submemu-section-title__icon`}>
+                <ArrowRight20 />
+              </div>
+            </SideNavLink>
+
+            {renderNavItem(item.megapanelContent.quickLinks.links)}
+          </SideNavMenu>
```

Comment:
This `<SideNavMenu>` and `<SideNavMenuItem>`/`<SideNavLink>` in it seems to work as a set of component, like "`<SideNavMenu>` with back/forward feature". If so, to better align to the Web Component version we'll develop in future, and better separate the render-from-data logic vs. render-UI logic, I suggest we create one React component, so the markup here will be like `<SideNavMenuWithBackForward>{renderNavItem(item.megapanelContent.quickLinks.links)}</SideNavMenuWithBackForward>`, if it makes sense.